examples/voxel_sdf/main.rs

```diff
@@ -0,0 +1,47 @@
+use dotrix::{camera, ecs::Mut, Camera, Dotrix, System, World};
+use dotrix_voxel::{Grid, Light, TexSdf, VoxelJumpFlood};
+use rand::Rng;
+
+fn main() {
+    Dotrix::application("Dotrix: Voxel SDF")
+        .with(System::from(startup))
+        .with(System::from(camera::control))
+        .with(dotrix_voxel::extension)
+        .run();
+}
+
+fn startup(mut camera: Mut<Camera>, mut world: Mut<World>) {
+    camera.target = [0., 0., 0.].into();
+    camera.distance = 30.0;
+    camera.tilt = 0.0;
+
+    let mut grid = Grid::default();
+
+    let dims = grid.dimensions;
+    let total_size: usize = dims.iter().fold(1usize, |acc, &item| acc * (item as usize));
+    let values: Vec<u8> = vec![0u8; total_size]
+        .iter()
+        .map(|_v| {
+            let chance: u8 = rand::thread_rng().gen();
+            if chance > 128 {
+                1
+            } else {
+                0
+            }
+        })
+        .collect();
+
+    grid = grid.with_values(values);
+    world.spawn(vec![(grid, VoxelJumpFlood::default(), TexSdf::default())]);
```

Comment:
[question]
Will it react on change of the grid in runtime?

Reply:
Not yet but it should not be hard to implement. All we need to do is delete the sdf texture and seed texture when changes occur.

Reply:
Live reload is now working checkout the randomize button in the demo